### ISSUE TITLE
--monitor-snapshots not catching missing snapshots / sanoid not creating missing snapshots

### ISSUE THREAD

**Issue by darkpixel** [jimsalterjrs/sanoid#562]
```
sanoid --monitor-snapshots
CRIT: tank/globalusers has no monthly snapshots at all!
root@usbvesdnas01:~# zfs list -rt snapshot -o name -s name -H tank/globalusers | grep -v frequently | grep -v hourly | grep -v daily
root@usbvesdnas01:~# 
```
It certainly appears there are no monthly snapshots...but there's also no 'weekly' snapshots and it's not alerting about that either.  Does it just alert on the first missing snapshot it finds and then quits searching?

**Comment by phreaker0** [comment]
@darkpixel as weekly snapshots are turned off by default I guess you just enabled the weekly snapshots bot did not set/enable the weekly monitoring settings, they default to not check anything as weekly are off by default:

```
# setting any value to 0 will keep the monitoring service from monitoring that snapshot type on that section at all.
....
weekly_warn = 0
weekly_crit = 0
...
```

**Comment by darkpixel** [comment]
Sorry--I should have updated this sooner.
My monitoring system started going nuts over this box after I posted the bug report.  It started saying there we no daily, hourly, frequently, etc...for various datasets.  On the next check it would report everything was fine.  Then different datasets would be missing stuff.

I finally stopped cron and made sure sanoid wasn't runningand tried running ```sanoid --force-update```.  It did its thing, and monitoring still was reporting stuff incorrectly and differently every time I ran ```sanoid --monitor-snapshots```.  It would list datasets missing 'hourly' snapshots that had 24 on it.

I ran ```sanoid --force-update``` again it threw a handful of Perl errors.

I finally rm'd ```/var/cache/sanoidsnapshots.txt``` and re-ran ```sanoid```.  It updated its cache, created a few new 'frequently' snapshots and everything went back to normal.  I'm kicking myself for not saving a copy of ```sanoidsnapshots.txt``` to see what the heck happened...but it's been fine now for about 3.5 hours.
